# https://trac.macports.org/ticket/74524
Date: Mon, 05 Oct 2026 11:05:48 -0000
#74524: openexr: build fails on 10.14 and earlier, due to aligned_alloc
Comment (by tomio-arisaka):

--- src/lib/OpenEXRCore/internal_zstd.c.orig	2026-10-02 13:36:03.000000000 +0900
+++ src/lib/OpenEXRCore/internal_zstd.c	2026-10-04 22:47:05.000000000 +0900
@@ -184,6 +184,19 @@
     return t_exr_zstd_tls_fast;
 }

+/* On macOS 10.13, aligned_alloc is not provided,
+   so use posix_memalign instead
+*/
+static void *aligned_alloc(size_t align, size_t size)
+{
+    void *ptr;
+
+    if (posix_memalign(&ptr, align, size) != 0)
+      return NULL; // error
+    else
+      return ptr;
+}
+
 /** Returns true on success (tls->shuffle_buf is valid and at least
  *  required_size bytes), false on failure (allocation failure or the
  *  overshoot/alignment arithmetic below would overflow size_t). On failure
